ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.cs

```diff
@@ -386,32 +389,41 @@ public override async Task<StreamState> ReceiveStream(StreamState state, Progres
       {
         RhinoDoc.ActiveDoc.Notes += "%%%" + commitLayerName; // give converter a way to access commit layer info
 
-      // create preview objects if they don't already exist
-      if (Preview.Count == 0)
-      {
-        // flatten the commit object to retrieve children objs
-        int count = 0;
-        Preview = FlattenCommitObject(commitObject, converter, progress, commitLayerName, ref count);
-
-        // convert
-        foreach (var previewObj in Preview)
+        // create preview objects if they don't already exist
+        if (Preview.Count == 0)
         {
-            var storedObj = StoredObjects[previewObj.OriginalId];
```

Comment:
This line was sometimes throwing an exception, as `previewObj.OrianlId` is not in `StoredObjects`

Previously (rc3), this lookup wouldn't happen until Convert was called. And wouldn't happen for paths where the object is not convertible.